packages/xstate-immer/test/immer.test.ts

```diff
@@ -100,40 +101,108 @@ describe('@xstate/immer', () => {
     expect(twoState.context.foo.bar.baz).toEqual([1, 2, 3, 0, 0]);
   });
 
-  it('should patch updates', () => {
+  it('should create updates', () => {
     const context = {
       foo: {
         bar: {
           baz: [1, 2, 3]
         }
       }
     };
-    const countMachine = Machine<typeof context>({
+
+    const bazUpdater = createUpdater<
+      typeof context,
+      ImmerUpdateEvent<'UPDATE_BAZ', number>
+    >('UPDATE_BAZ', (ctx, input) => {
+      ctx.foo.bar.baz.push(input);
+    });
+
+    const countMachine = createMachine<typeof context>({
       id: 'count',
       context,
       initial: 'active',
       states: {
         active: {
           on: {
-            UPDATE_BAZ: {
-              actions: assignPatch()
+            [bazUpdater.type]: {
+              actions: bazUpdater.assign
             }
           }
         }
       }
     });
 
     const zeroState = countMachine.initialState;
-    const somePatchEvent = patchEvent(
-      'UPDATE_BAZ',
-      countMachine.initialState.context,
-      (ctx) => {
-        ctx.foo.bar.baz.push(4);
-      }
-    );
-    const twoState = countMachine.transition(zeroState, somePatchEvent);
+
+    const twoState = countMachine.transition(zeroState, bazUpdater(4));
 
     expect(zeroState.context.foo.bar.baz).toEqual([1, 2, 3]);
     expect(twoState.context.foo.bar.baz).toEqual([1, 2, 3, 4]);
   });
+
+  it('should create updates (form example)', (done) => {
+    const context = {
+      name: '',
+      age: undefined as number | undefined
+    };
+
+    const nameUpdater = createUpdater<
+      typeof context,
+      ImmerUpdateEvent<'UPDATE_NAME', string>
+    >('UPDATE_NAME', (ctx, input) => {
+      ctx.name = input;
+    });
+
+    const ageUpdater = createUpdater<
+      typeof context,
+      ImmerUpdateEvent<'UPDATE_AGE', number>
+    >('UPDATE_AGE', (ctx, input) => {
+      ctx.age = input;
+    });
+
+    type FormEvent =
+      | ReturnType<typeof nameUpdater>
+      | ReturnType<typeof ageUpdater>
+      | {
+          type: 'SUBMIT';
+        };
+
+    const formMachine = createMachine<typeof context, FormEvent>({
+      initial: 'editing',
+      context,
+      states: {
+        editing: {
+          on: {
+            [nameUpdater.type]: { actions: nameUpdater.assign },
```

Comment:
probably we could use util for this, seems repetitive - can be done later though if we ain't 100% sure how to name such a util right now

Reply:
It's a lot less verbose than the alternative. I'm in favor of this for keeping it explicit, and for not limiting the use-cases - we might want there to be multiple actions, or custom guards, etc.

Reply:
the base API is good - we definitely dont want to limit use cases, I've just had in mind a helper API for producing those simple transitions which would cover most basic use cases. It's also easily implementable in the userland, so probably at the end of the day it's best to leave it to userland.